iOS Client/ArticlesController/MBArticlesViewController.swift

```diff
@@ -47,34 +49,42 @@ class MBArticlesViewController: UIViewController, UITableViewDelegate, UITableVi
         
         tableView.rowHeight = UITableViewAutomaticDimension
         tableView.estimatedRowHeight = CGFloat(MBConstants.ARTICLE_TABLEVIEWCELL_HEIGHT)
+        tableView.refreshControl = refreshControl
+        refreshControl.addTarget(self, action: #selector(refreshTableView(_:)), for: .valueChanged)
+        refreshControl.tintColor = UIColor(red:235.0/255.0, green:96.0/255.0, blue:93.0/255.0, alpha:1.0)
+        refreshControl.attributedTitle = NSAttributedString(string: "Fetching Articles ...", attributes: nil)
         
-        let oneWeekAgoTimestamp = Date().timeIntervalSinceReferenceDate - MBConstants.SECONDS_IN_A_WEEK
+        let oneDayAgoTimestamp = Date().timeIntervalSinceReferenceDate - MBConstants.SECONDS_IN_A_DAY
         let lastUpdateTimestamp = UserDefaults.standard.double(forKey: MBConstants.DEFAULTS_KEY_ARTICLE_UPDATE_TIMESTAMP)
         
-        if lastUpdateTimestamp > oneWeekAgoTimestamp {
-            let articles = articlesStore.getArticles(managedObjectContext: self.managedObjectContext)
-            
-            if articles.count > 0 {
-                MBStore.sharedStore.dispatch(LoadedArticles(articles: .loaded(data: articles)))
-                return
-            }
-        }
+        let articles = articlesStore.getArticles(managedObjectContext: self.managedObjectContext)
+        MBStore.sharedStore.dispatch(LoadedArticles(articles: .loaded(data: articles)))
         
-        downloadArticleData()
+        if lastUpdateTimestamp < oneDayAgoTimestamp || articles.count == 0 {
```

Comment:
Is there a reason not to refresh every time viewDidLoad is called. Since this will only really be called if the app is killed and rebooted I would think people would want to see all the latest articles when that happened.

Reply:
agreed